File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/StockAdjustment.js

// Define collection and schema for StockAdjustment
export interface StockAdjustment {
    adjustmentNumber:
	type : string
    reason:
	type : string
    adjustmentDate:
	type : Date
    Warehouse:
	type : Schema.Types.ObjectId
    Lines:
 	type : [{ type: Schema.Types.ObjectId, ref: 'StockAdjustmentLine' }]
    Transactions:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventoryTransaction' }]
    AdjustmentType:
 	type : String
    Status:
 	type : String
#
    collection: 'stockAdjustments'
}

File: evidence/generated-systems/rubyonrails/inventory-on-rails/inventoryonrails/app/models/stockadjustment.rb
class StockAdjustment < ApplicationRecord
  enum AdjustmentType: [:Increase, :Decrease, :Reclassification]
  enum Status: [:Draft, :Approved, :Posted, :Cancelled]


  has_many :Warehouse, class_name: 'Warehouse'
  has_many :Lines, class_name: 'StockAdjustmentLine'
  has_many :Transactions, class_name: 'InventoryTransaction'

end
